src/libstd/sys/unix/fs.rs

```diff
@@ -459,9 +459,18 @@ pub fn rmdir(p: &Path) -> io::Result<()> {
 }
 
 pub fn readlink(p: &Path) -> io::Result<PathBuf> {
+    #[cfg(not(target_libc = "newlib"))]
```

Comment:
Ah ok, I'd prefer to see everything all at once, so feel free to just lump everything in here!